src/scanner/mod.rs

```diff
@@ -130,8 +134,22 @@ impl Scanner {
     ///
     /// Note: `self` must contain `self.ip`.
     async fn scan_socket(&self, socket: SocketAddr) -> io::Result<SocketAddr> {
-        let tries = self.tries.get();
+        if self.udp {
+            let waits = vec![0, 51, 107, 313];
```

Comment:
question: can I check where these values have come from?

Are they particularly meaningful? And worth documenting?

Reply:
No, they were 50, 100, 300.

bee-san suggested 
> Suggestion: Make the times less even? A firewall would pick up on even timing. Ideally we'd use random times 
> but it's slow and should be in a --quiet option (one for another PR)
> Just doing things like 0, 51, 107, 313 etc would be good :)

Would've never thought of this myself

Reply:
@PsypherPunk  Because of the way UDP works we don't know how long to wait to assume failure. Normally TCP would always get back to us, and if it doesn't we wait 1 second I think.

We need to tell UDP how long to wait for before moving on, I was thinking if we send multiple UDP packets at even timestamps the firewall would easily pick up on it. Uneven doesn't help _so_ much but it's slightly better than nothing.

Something I realised is we have `0` here. Reading the code:

https://github.com/RustScan/RustScan/blob/52bfcf048db5e10bcdd887abb350c85606331760/src/scanner/mod.rs#L258

Does this imply we wait `0` seconds before assuming it's closed?
And thinking more about it do we need a vector of timeouts here at all? 

The normal TCP scan waits 1.5 seconds:
https://github.com/RustScan/RustScan/blob/52bfcf048db5e10bcdd887abb350c85606331760/src/input.rs#L116

I think you should actually parse the `timeout` input down here and use that instead? That way in the future we can update this to a random timeout if we want to :)

Sorry, only just caught this 🙈

Reply:
Yes actually thinking about this more, there's other functions which use the timeout given at the argparsing level so I think we should not include this vector and just parse the timeout from the argparsing part (like the other tcp scanner does)

and I think a nice lil feature would be some code that can alter that timeout more in depth, like randomised timeouts